xdsl/irdl/attributes.py

```diff
@@ -256,8 +262,30 @@ def irdl_attr_definition(cls: TypeAttributeInvT) -> TypeAttributeInvT:
     )
 
 
+IRDLGenericAttributeConstraint: TypeAlias = (
+    GenericAttrConstraint[AttributeInvT]
+    | Attribute
+    | type[AttributeInvT]
+    | "TypeForm[AttributeInvT]"
+    | ConstraintVar
+    | TypeVar
```

Comment:
Adding `typing_extensions.TypeVar`  here would probably fix your previous PR.

Reply:
If that's okay, I'll not fix the last PR CI, and merge both commits at once.
Otherwise there are a lot of places to add `typing_extensions.TypeVar` that is going to be replaced by this commit anyway!

Reply:
Maybe we should just use `typing_extensions.TypeVar` everywhere

Reply:
That's something I'm not 100% sure, because that's asking everyone to use typing_extensions.
I think we can just support both, I can add support in another PR

Reply:
Ah, I see what you meant by your message now